harness/features/track.local.test.ts

```diff
@@ -0,0 +1,268 @@
+import {
+    getConnectionStringForProxy,
+    forEachSDK,
+    describeIf,
+    createClient,
+    createUser,
+    wait,
+    callOnClientInitialized,
+    callTrack,
+    waitForRequest
+} from '../helpers'
+import { Capabilities, SDKCapabilities } from '../types'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../nock'
+import { config } from '../mockData/index'
+import nock from 'nock'
+
+jest.setTimeout(15000)
+
+const scope = getServerScope()
+
+describe('Track Tests - Local', () => {
+    const validUserId = 'user1'
+    forEachSDK((name) => {
+        let url: string
+        const capabilities: string[] = SDKCapabilities[name]
+        const clientId: string = uuidv4()
+        const mockServerUrl
+            = `http://${process.env.DOCKER_HOST_IP ?? 'host.docker.internal'}:${global.__MOCK_SERVER_PORT__}`
+        const eventFlushIntervalMS = 1000
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)
+
+            const sdkKey = `dvc_server_${clientId}`
+
+            scope
+                .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                .reply(200, config)
+
+            scope.post(`/client/${clientId}`).reply((uri, body) => {
+                if (typeof body === 'object'
+                    && body.message.includes('onClientInitialized was invoked'))
+                    return [200]
+            })
+
+            await createClient(url, clientId, sdkKey, {
+                baseURLOverride: `${mockServerUrl}/client/${clientId}`,
+                eventFlushIntervalMS: eventFlushIntervalMS, logLevel: 'debug', configPollingIntervalMS: 1000 * 60
+            })
+            await callOnClientInitialized(clientId, url, `${mockServerUrl}/client/${clientId}`)
```

Comment:
I think the problem here is that nothing is actually waiting for the client to finish initializing. This call is just "registering" the callback handler but nothing is actually waiting for that callback to be received. That's probably why you're seeing inconsistent behaviour when running the tests since there's basically a race condition.

The TestClient helper fixes this by changing the way it does onClientInitialized to just send a request with `isAsync` true. You could fix it here by either switching to use the TestClient helper or just update this existing `callOnClientInitialized` method to send "isAsync" true instead of trying to register a callback.

Reply:
this should fix the check you had to comment out below

Reply:
makes sense, could just use waitForRequest with the `onClientInitialized interceptor` and fix this so that the client is initialized before any tests run.

Reply:
you could, but its a bit of a roundabout approach since passing `isAsync` on the onClientInitialized request will do the same thing